service/src/config.rs

```diff
@@ -327,13 +330,23 @@ impl From<&ArgMatches<'_>> for RunConfig {
 				.to_string()
 		});
 
+		let shielding_target = m.value_of("shielding-target").map(|i| match i {
+			"integritee" => ParentchainId::Integritee,
+			"target_a" => ParentchainId::TargetA,
+			"target_b" => ParentchainId::TargetB,
+			_ => panic!(
+				"failed to parse shielding-target: {} must be one of integritee|target_a|target_b",
+				i
+			),
+		});
```

Comment:
This is what I did for another project of mine:

```rust
impl std::str::FromStr for RunConfig {
	type Err = String;

	fn from_str(s: &str) -> Result<Self, Self::Err> {
		match s {
			"feedback-only" => Ok(Self::FeedbackOnly),
			"print-only" => Ok(Self::PrintOnly),
			"send-tx" => Ok(Self::SendTx),
			_ => Err(format!("Unknown run config: {s}")),
		}
	}
}
```

The good thing is, clap v4 will use the `FromStr` trait implementation behind the scenes automatically. ;)

Reply:
nice. as the benefits of this would only show in clapv4 I'd suggest to leave this for the sake of *time-to-market*